services/offline/src/lib/cacheable-section-state.js

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import {
+    createStore,
+    useGlobalState,
+    useGlobalStateMutation,
+    GlobalStateProvider,
+} from './global-state-service'
+import { useOfflineInterface } from './offline-interface'
+
+// Functions in here use the global state service to manage cacheable section
+// state in a performant way
+
+/**
+ * Create a store for Cacheable Section state.
+ * Expected to be used in app adapter
+ */
+export function createCacheableSectionStore() {
+    const initialState = { recordingStates: {}, cachedSections: {} }
+    return createStore(initialState)
+}
+
+/**
+ * Provides context for a global state context which will track cached
+ * sections' status and cacheable sections' recording states, which will
+ * determine how that component will render. The provider will be a part of
+ * the OfflineProvider.
+ */
+export function CacheableSectionProvider({ children, store }) {
+    const offlineInterface = useOfflineInterface()
+
+    // On load, get sections and add to store
+    React.useEffect(() => {
+        offlineInterface.getCachedSections().then(sections => {
+            const newSections = sections.reduce(
```

Comment:
`sectionsById` or `lastUpdatedById` might be clearer than `newSections` here

Reply:
Agreed 👍